apps/prairielearn/assets/scripts/lib/countdown.ts

```diff
@@ -88,15 +96,16 @@ export function setupCountdown(options: {
     }
   }
 
-  function updateServerRemainingMS() {
+  function updateServerRemainingMS(lastRemainingMS: number) {
     if (options.serverUpdateURL) {
       fetch(options.serverUpdateURL)
         .then(async (response) => {
           handleServerResponseRemainingMS(await response.json());
         })
         .catch((err) => {
           console.log('Error retrieving remaining time', err);
-          options.onServerUpdateFail?.();
+          options.onServerUpdateFail?.(lastRemainingMS);
+          displayCountdown();
```

Comment:
The addition of `displayCountdown` here has a pretty nasty effect if the server is temporarily unavailable: `updateServerRemainingMS` and `displayCountdown` will recursively call each other very very fast. You can reproduce this by stopping the PL server when 30 seconds are left. Watch your browser dev tools: you'll see tens (hundreds?) of requests per second to the `time_remaining` endpoint.

Reply:
I agree that's not great behavior, but the best alternative I can think of that does not involve hard-reloading the page is to introduce a delay (something like 10 seconds?) that prevents the repeated server requests.

Reply:
This should be addressed now. It seems that this issue only occurred when the server failed after the timer expired, and now there's an extra check (similar to the handling of the successful case) where further server updates are disabled in this case.